src/_pytest/capture.py

```diff
@@ -855,7 +855,7 @@ def capsysbinary(request: SubRequest):
 
 
 @pytest.fixture
-def capfd(request: SubRequest):
+def capfd(request: SubRequest) -> Generator[CaptureFixture, None, None]:
```

Comment:
I think it's possible using [generic type aliases](https://mypy.readthedocs.io/en/stable/generics.html#generic-type-aliases), however I prefer not to because:

- It adds indirection -- reader needs to understand the alias, while `Generator` is a common concept.
- Requires extra import.

Reply:
> It adds indirection -- reader needs to understand the alias, while Generator is a common concept.

I see this the other way around: it abstracts and makes it clear this is a "yield fixture", and not a common generator. 😁 

But I will defer to your decision here of course!